lib/matplotlib/widgets.py

```diff
@@ -404,11 +419,33 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
         self.val = valinit
         self.valinit = valinit
         if orientation == 'vertical':
-            self.poly = ax.axhspan(valmin, valinit, 0, 1, **kwargs)
-            self.hline = ax.axhline(valinit, 0, 1, color=initcolor, lw=1)
+            self.track = Rectangle(
+                (.25, 0), .5, 1,
+                transform=ax.transAxes,
+                facecolor=track_color
+            )
+            ax.add_patch(self.track)
+            self.poly = ax.axhspan(valmin, valinit, .25, .75, **kwargs)
+            self.hline = ax.axhline(valinit, .25, .75, color=initcolor, lw=1)
```

Comment:
```suggestion
            self.hline = ax.axhline(valinit, .2, .8, color=initcolor, lw=1)
```
The init line extends one pixel more to the bottom (or right for vertical sliders), which looks a bit awkward.

I suspect that the effect comes from different to-pixel mappings of lines and rectangles (lines render including both end points; i.e. two lines ending in the same point would draw on top of each other. But rectangles are designed to not overlap if they are side-by-side). - This is just speculation, but I've checked and excluded some other possible reasons (capstyle and snapping).

The cheap solution to get away with a reasonably looking init marker is to let it extend slightly on both sides. The pixel difference is still there, but 2 vs. 3 is less noticeable than 0 vs. 1.

Reply:
I agree that this simple solution is the way to go. Though I actually feel it is more noticeable in the horizontal slider in your second screenshot. Maybe this is due to your screenshot program but it looks as though on the vertical slider it is 2 vs 3 (or even the same):
![image](https://user-images.githubusercontent.com/10111092/104862387-35f4b100-5901-11eb-89fa-259712088cf9.png)

whereas on the horizontal slider:
![image](https://user-images.githubusercontent.com/10111092/104862459-4e64cb80-5901-11eb-8f18-651ce3fc9d73.png)